src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -0,0 +1,171 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+import com.google.gson.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.QueryResultList;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.User;
```

Comment:
We're getting a little tripped up on UserService here. To give you some more background, AppEngine's UserService and Google Sign In are essentially different services providing the same functionality - you'd use one or the other, but not both. At least not both in the way you're currently using them.

Here's my understanding of how you'd integrate Google Sign In with this project:

- Follow the [Google Sign In example](https://developers.google.com/identity/sign-in/web/sign-in#get_profile_information), as you've already done here.
- The example only gives you information about the signed in user in Javascript (client-side) context. Your backend (UserServlet.java's UserService) is unaware of the user. When running this PR, you'll notice that calls to GET /user will respond with a 500 - a NullPointerException is raised in UserServlet.java:77 since the userService in `userService.getCurrentUser().getUserId()` doesn't actually know about the user. Remember that only your JS code knows about the user. AppEngine's UserService, a completely different service running in your servlet, is unaware of any user.
- Since we're using Google Sign In, we have to rely on Google Sign In to provide data to the backend on which user is currently signed in. To do so, read and follow https://developers.google.com/identity/sign-in/web/backend-auth.
- In https://developers.google.com/identity/sign-in/web/backend-auth#using-a-google-api-client-library, you'll notice that once `GoogleIdTokenVerifier` has verified the user token, you can use the user id as provided in 
```
GoogleIdToken idToken = verifier.verify(idTokenString);
if (idToken != null) {
  Payload payload = idToken.getPayload();

  // Print user identifier
  String userId = payload.getSubject();
```
to retrieve an id suitable for use throughout your backend. Concretely, you should add a `string googleUserId` field to UserInfo and a corresponding property to the underlying Entity. Note that since userId is a string, you can't use it as your Entity.Key.
- [As outlined in the example](https://developers.google.com/identity/sign-in/web/backend-auth#send-the-id-token-to-your-server), for all GET/PUT/PATCH requests to the /user endpoint, you'll need to provide a `id_token` request parameter provided by `auth2.currentUser.get().getAuthResponse().id_token;` in your JS to identify the current user.


Implementation note: when you instantiate `GoogleIdTokenVerifier`, you'll want to use AppEngine's `com.google.api.client.extensions.appengine.http.UrlFetchTransport` (as documented [here](https://googleapis.dev/java/google-http-client/1.34.2/com/google/api/client/http/HttpTransport.html?is-external=true)) and [GsonFactory](https://googleapis.dev/java/google-http-client/1.34.2/com/google/api/client/json/gson/GsonFactory.html)